pynecone/components/graphing/victory.py

```diff
@@ -0,0 +1,567 @@
+        ValueError: If y is provided and x, y are not the same length.
+        ValueError: If y is not provided and min, max, median, q1, and q3 are not provided.
+        ValueError: If y is not provided and x, min, max, median, q1, and q3 are not the same length.
+    """
+    data = []
+    if x is None:
+        raise ValueError("x must be specified")
+
+    if y is not None:
+        return format_xy(x, y)
+
+    else:
+        if min_ is None or max_ is None or median is None or q1 is None or q3 is None:
+            raise ValueError(
+                "min, max, median, q1, and q3 must be specified if y is not provided"
+            )
+        elif (
+            len(x) != len(min_)
+            or len(x) != len(max_)
+            or len(x) != len(median)
+            or len(x) != len(q1)
+            or len(x) != len(q3)
+        ):
+            raise ValueError(
+                "x, min, max, median, q1, and q3 must be the same length and specified if y is not provided"
+            )
+        for i in range(len(x)):
+            row = {}
+            if x is not None:
+                row["x"] = x[i]
+            if min_ is not None:
+                row["min"] = min_[i]
+            if max_ is not None:
+                row["max"] = max_[i]
+            if median is not None:
+                row["median"] = median[i]
+            if q1 is not None:
+                row["q1"] = q1[i]
+            if q3 is not None:
+                row["q3"] = q3[i]
+            data.append(row)
+    return data
+
+
+def format_histogram(x: List) -> List:
+    """Format histogram data.
+
+    Args:
+        x: The x values.
+
+    Returns:
+        The formatted data.
+
+    Raises:
+        ValueError: If x is not provided.
+    """
+    if x is None:
+        raise ValueError("x must be specified")
+
+    return [{"x": x[i]} for i in range(len(x))]
+
+
+def format_pie(x: List, y: List, label: Optional[List] = None) -> List:
+    """Format pie data.
+
+    Args:
+        x: The x values.
+        y: The y values.
+        label: The label values.
+
+    Returns:
+        The formatted data.
+
+    Raises:
+        ValueError: If x is not provided.
+        ValueError: If x and y are not the same length.
+        ValueError: If x, y, and label are not the same length.
+    """
+    if x is None:
+        raise ValueError("x must be specified")
+
+    if label is None:
+        return format_xy(x, y)
+    else:
+        if len(x) != len(y) or len(x) != len(label):
+            raise ValueError("x, y, and label must be the same length")
+        return [{"x": x[i], "y": y[i], "label": label[i]} for i in range(len(x))]
+
+
+def format_voronoi(x: List, y: List) -> List:
+    """Format voronoi data.
+
+    Args:
+        x: The x values.
+        y: The y values.
+
+    Returns:
+        The formatted data.
+
+    Raises:
+        ValueError: If x or y is not provided.
+        ValueError: If x and y are not the same length.
+    """
+    data = []
```

Comment:
Don't need this variable. Also, I think the `None` check can be moved into the `format_xy` function